js/src/otel.ts

```diff
@@ -305,11 +305,41 @@ export class BraintrustSpanProcessor {
         ...options.headers,
       };
 
-      exporter = new OTLPTraceExporter({
+      const baseExporter = new OTLPTraceExporter({
         url: new URL("otel/v1/traces", apiUrl).href,
         headers,
       });
+
+      exporter = new Proxy(baseExporter, {
+        get(target, prop, receiver) {
+          // If the code is trying to access the 'export' method, return our patched version.
+          if (prop === "export") {
+            return function (
+              spans: any[],
+              resultCallback: (result: any) => void,
+            ) {
+              // This patch handles OTel version mismatches
+              const fixedSpans = spans.map((span: any) => {
+                if (!span.instrumentationScope && span.instrumentationLibrary) {
+                  span.instrumentationScope = span.instrumentationLibrary;
+                }
+                return span;
+              });
+
+              // Call the original export method with the fixed spans.
+              return Reflect.apply(target.export, target, [
+                fixedSpans,
+                resultCallback,
+              ]);
+            };
+          }
+
+          // For any other property, pass the access through to the original object.
+          return Reflect.get(target, prop, receiver);
+        },
+      });
     } catch (error) {
+      console.error(error);
```

Comment:
#nit we can make this better, but I tripped over this where any problem within this code will cause the same "make sure ... is installed" was was _very_ misleading